code/model/editableformfields/EditableFormField.php

```diff
@@ -984,4 +1024,73 @@ public function EffectiveDisplayRules()
         }
         return $this->DisplayRules();
     }
+
+	/**
+	 * @return array|null
+	 */
+	public function formatDisplayRules()
+	{
+		$field = $this;
+		$holderSelector = $field->getSelectorOnly();
+		$result = array(
+			'targetFieldID' => $holderSelector,
+			'conjunction'   => $field->DisplayRulesConjunctionNice(),
+			'selectors'     => array(),
+			'events'        => array(),
+			'operations'    => array(),
+			'initialState'  => $this->ShowOnLoadNice(),
+			'view'  => array(),
+			'opposite'  => array(),
+		);
+		// Check for field dependencies / default
+		/** @var EditableCustomRule $rule */
+		foreach($field->EffectiveDisplayRules() as $rule)
+		{
+			echo $rule->FieldValue;
+			// Get the field which is effected
+			/** @var EditableFormField $formFieldWatch */
+			$formFieldWatch = EditableFormField::get()->byId($rule->ConditionFieldID);
+			// Skip deleted fields
+			if(!$formFieldWatch)
+			{
+				echo 'haha';
+				continue;
+			}
+			$fieldToWatch = $formFieldWatch->getSelectorFieldOnly();
+
+			$expression = $rule->BuildExpression();
+			if(!in_array($fieldToWatch, $result['selectors']))
+			{
+				$result['selectors'][] = $fieldToWatch;
+			}
+			if(!in_array($expression['event'], $result['events']))
+			{
+				$result['events'][] = $expression['event'];
+			}
+			$result['operations'][] = $expression['operation'];
+
+			//View/Show should read
+			$result['view'] = $rule->ToggleDisplayText($result['initialState']);
+			$result['opposite'] = $rule->ToggleDisplayText($result['view']);
+		}
+
+		return (count($result['selectors'])) ? $result : NULL;
+	}
+
+	public function DisplayRulesConjunctionNice()
+	{
+		return (strtolower($this->DisplayRulesConjunction) === 'or') ? '||' : '&&';
+	}
+
+	public function ShowOnLoadNice(){
+		return  ($this->ShowOnLoad)?'show':'hide';
+	}
+
+	public function isCheckBoxField(){
+		return false;
+	}
+
+	public function isRadioField(){
```

Comment:
I've opted for base method overriding behaviour here instead, where EditableFormField->isRadioField() gets overridden in in EditableRadioField->isRadioField()